api/envoy/api/v2/cds.proto

```diff
@@ -632,6 +632,25 @@ message Cluster {
   // If this flag is not set to true, Envoy will wait until the hosts fail active health
   // checking before removing it from the cluster.
   bool drain_connections_on_host_removal = 32;
+
+  // New mechanism for LB policy configuration. If set and understood
+  // by the client, replaces the lb_policy field, the common_lb_config
+  // field, and the fields in the lb_config oneof. For backward
+  // compatibility, management servers should populate both sets of fields.
+  //
+  // Every LB policy defined via this mechanism will be identified via a
+  // unique name. If the policy that needs configuration parameters, it
+  // musts define a message for its own configuration, which will be stored
+  // in the config field. The name of the policy will tell clients
+  // which type of message they should expect to see in the config field.
+  message IntraLocalityLoadBalancingPolicy {
+    string name = 1;                // Required. The name of the LB policy.
+    google.protobuf.Any config = 2; // Optional. Config for the LB policy.
+  }
+  // Each client will iterate over the list in order and stop at the first
+  // policy that it supports. This provides a mechanism for starting to
+  // use new LB policies that are not yet supported by all clients.
```

Comment:
I've gone ahead and done this.

Note that for the three policies that already had custom config message types (least_request, ring_hash, and original_dst), I've added a `CommonLbConfig` field in their custom config messages, so that they have a way to include the common configuration.  For the other policies, I just listed `CommonLbConfig` as their config message type.

It's not clear what we need to do about the `lb_subset_config` field, since I don't know how that interacts with the LB policy in Envoy.  Is it the case that the subsetting functionality can be used with any arbitrary LB policy (e.g., Envoy divides the hosts into subsets and then creates a separate LB policy for each subset)?  If so, that field can probably stay as-is.  Or is it the case that the LB policy must be aware of the subsetting config?  In that case, it might make more sense to move the `lb_subset_config` field into the `CommonLbConfig` message, so that it can be used with Envoy's built-in policies but cannot be mistakenly configured with a third-party policy that does not support this functionality.